src/mapgen_functions.h

```diff
@@ -46,6 +46,12 @@ struct mapgendata
   bool is_groundcover(const ter_id iid ) const;
 };
 
+/**
+ * Calculates the coordinates of a rotated point.
+ * Should match the `mapgen_*` rotation.
+ */
```

Comment:
It's rotation around `[11.5, 11.5]` by `90 * turn` degrees counterclockwise.

Though I'm not sure if it would be the best location for the description. After all this rotation is applied to all mapgen entries that have rotation.